backend/src/index.js

```diff
@@ -285,6 +319,24 @@ async function main() {
   };
   setTimeout(regularCheckNodeStatus, 0);
 
+  // Periodic check of validator's fee and delegators
+  const regularCheckValidatorsExtraInfo = async (validators) => {
```

Comment:
Use a dedicated `setTimeout` for this `regularCheckValidatorsExtraInfo`, and don't call it from `regularCheckNodeStatus`. Fetch the validators (current validators and current proposals) inside this regular job, so you do not need to call it from `regularCheckNodeStatus`.